packages/lune/src/builder/rolldown/solid.ts

```diff
@@ -0,0 +1,48 @@
+import { transformAsync, type TransformOptions } from "@babel/core";
+import ts from "@babel/preset-typescript";
+import solid from "babel-preset-solid";
+import type { Plugin } from "rolldown";
+
+function getExtension(filename: string): string {
+  const index = filename.lastIndexOf(".");
+  return index < 0 ? "" : filename.substring(index).replace(/\?.+$/, "");
+}
+
+export const SolidPlugin = (): Plugin => {
+  let projectRoot = process.cwd();
+  return {
+    name: "lune:rolldown:solid-js",
+    transform: {
+      // Handle only .tsx and .jsx files
+      filter: { id: /(\.tsx|\.jsx)$/ },
+      handler: async (source, id) => {
+        const currentFileExtension = getExtension(id);
+
+        const solidOptions = { generate: "dom", hydratable: true };
+
+        id = id.replace(/\?.+$/, "");
+
+        const babelOptions: TransformOptions = {
+          babelrc: false,
+          configFile: false,
+          root: projectRoot,
+          filename: id,
+          sourceFileName: id,
+          presets: [[solid, { ...solidOptions }]],
+          plugins: [],
+          sourceMaps: true,
+        };
+
+        const shouldBeProcessedWithTypescript = currentFileExtension === ".tsx";
+
+        if (shouldBeProcessedWithTypescript) {
+          babelOptions.presets!.push([ts, {}]);
```

Comment:
Somewhat concerned about the idea that we call out to Babel to process typescript, as I *assume* whatever is being used internally by rolldown for `.ts` files is much faster.

Reply:
Okay this seems infeasible, as we would need to somehow do the tsx compilation later in the pipeline which I don't think is possible.